languages/ruby/lib/oso/polar/polar.rb

```diff
@@ -43,11 +46,31 @@ def clear
       # @param name [String]
       # @raise [PolarFileExtensionError] if provided filename has invalid extension.
       # @raise [PolarFileNotFoundError] if provided filename does not exist.
-      def load_file(name)
+      def load_file(name) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         raise PolarFileExtensionError unless ['.pol', '.polar'].include? File.extname(name)
-        raise PolarFileNotFoundError, name unless File.file?(name)
 
-        load_queue << name
+        begin
+          file = File.open(name)
+          file_data = file.read
+          hash = Digest::MD5.hexdigest(file_data)
+          file.close
+        rescue Errno::ENOENT
+          raise PolarFileNotFoundError, name unless File.file?(name)
+        end
+
+        if loaded_names.key?(name)
+          raise RepeatLoadError, "File #{name} has already been loaded." if loaded_names[name] == hash
```

Comment:
nit: I think these two errors should be different types. E.g., in the new JS lib I called 'em:
```ts
export class PolarFileAlreadyLoadedError extends PolarError {
  constructor(file: string) {
    super(`File '${file}' already loaded.`);
    Object.setPrototypeOf(this, PolarFileAlreadyLoadedError.prototype);
  }
}

export class PolarFileContentsChangedError extends PolarError {
  constructor(file: string) {
    super(`File '${file}' already loaded, but contents have changed.`);
    Object.setPrototypeOf(this, PolarFileContentsChangedError.prototype);
  }
}
```
fine w/ whatever names you decide on, but I think they should be different in case end users want to handle them differently w/o string matching on the message.